### PR TITLE
gh-1326 AddGraph shallowClone added missing graphAuths + tests.

### PR DESCRIPTION
**Pull request by GCHQDeveloper404**

### CODE REVIEW

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/AddGraph.java [hunk 1]
```diff
@@ -178,7 +179,7 @@ public Builder parentSchemaIds(final List<String> parentSchemaIds) {
             return _self();
         }
 
-        public Builder graphAuths(final String... graphAuths) {
+        public Builder setGraphAuths(final String... graphAuths) {
```
Comment (p013570) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/AddGraph.java]: This is now inconsistent with FederatedAccessHook.Builder.graphAuths. Perhaps just rename it back to graphAuths.

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/AddGraph.java [hunk 2]
```diff
@@ -89,6 +89,7 @@ public AddGraph shallowClone() throws CloneFailedException {
                 .parentSchemaIds(parentSchemaIds)
                 .parentPropertiesId(parentPropertiesId)
                 .options(options)
+                .setGraphAuths(graphAuths.toArray(new String[graphAuths.size()]))
```
Comment (p013570) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/AddGraph.java]: This logic now means the test FederatedStoreTest.shouldAddGraphIdWithAuths fails.

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/AddGraph.java (hunk 1 above)
Comment (GCHQDeveloper404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/AddGraph.java]: My reasoning was
```
.graphAuths("a").graphAuths("b").graphAuths("c").graphAuths("d")
```
Doesn't append, it overwrites. maybe I should try and change how the method is implemented?

Comment (GCHQDeveloper404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/AddGraph.java]: Undoing after our discussion.

**Pull request merged**